src/VisualStudio/Core/Def/Implementation/FindReferences/VisualStudioDefinitionsAndReferencesFactory.cs

```diff
@@ -102,10 +103,19 @@ private class ExternalDefinitionItem : DefinitionItem
                 _charOffset = charOffset;
             }
 
-            public override bool CanNavigateTo(Workspace workspace, CancellationToken cancellationToken) => true;
+            public override Task<bool> CanNavigateToAsync(Workspace workspace, CancellationToken cancellationToken)
+                => SpecializedTasks.True;
 
+            public override Task<bool> TryNavigateToAsync(Workspace workspace, bool showInPreviewTab, bool activateTab, CancellationToken cancellationToken)
+                => Task.FromResult(TryOpenFile() && TryNavigateToPosition());
```

Comment:
❗ This needs to switch to the main thread before calling `TryOpenFile()` or `TryNavigateToPosition()`

Reply:
indeed!